#!/usr/bin/env python

PACKAGE="end_row_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("start", bool_t, 0, "Start navigation:", False)
gen.add("p_gain",            double_t, 0,  "p gain: ", 3.0,    0.0, 100.0)
gen.add("i_gain",            double_t, 0,  "i gain: ", 0.0,    0.0, 0.5)
gen.add("d_gain",            double_t, 0,  "d gain: ", 0,    0.0, 0.5)

gen.add("use_speed_control", bool_t,   0, "use_speed_control: ",   True)
gen.add("speed_low",         double_t, 0, "minimal drive speed: ", 0.1,     0.0, 20.0)
gen.add("speed_high",        double_t, 0, "max drive speed: ",     1.0,     0.0, 20.0)

gen.add("correction_angle",        double_t, 0, "correction angle: ",     45,     0.0, 360.0)

gen.add("corridors_to_skip",        int_t, 0, "corridors to skip: ",     0,     0, 1000)
gen.add("correction_speed",        double_t, 0, "correction speed: ",     0.25,     0.0, 5.0)

gen.add("pole_radius",        double_t, 0, "pole radius: ",     1.0,     0.1, 5.0)

gen.add("use_barycenter_as_point", bool_t,   0, "use barycenter to find the segment: ",   False)


exit(gen.generate(PACKAGE, "end_row_param", "EndRowParam"))
